docs/shared/tutorials/npm-workspaces.md

```diff
@@ -0,0 +1,353 @@
+
+Now let's change where the `build` results are output to in the `vite.config.ts` file.
+
+```ts {% fileName="apps/demo/vite.config.ts" highlightLines=[7-9] %}
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
+  build: {
+    outDir: '../../dist/demo',
+  },
+});
+```
+
+Now if you look at project details view again, you'll see that the `outputs` property for Nx's caching has been updated to stay in sync with the setting in the `vite.config.ts` file.
+
+You can also add the `@nx/eslint` plugin to see how it infers `lint` tasks based on the ESLint configuration files.
+
+## Summary
+
+After following this tutorial, the repository is still using all the same tools to run tasks, but now Nx runs those tasks in a smarter way. The tasks are efficiently cached so that there is no repeated work and the cache configuration settings are automatically synced with your tooling configuration files by Nx plugins. Also, any task dependencies are automatically executed whenever needed because we configured task pipelines for the projects.
+
+The final task graph for `demo` app's `build` task looks like this:
+
+{% graph height="200px" title="Build Task Pipeline" type="task" jsonFile="shared/tutorials/npm-workspaces-build-tasks2.json" %}
+{% /graph %}
+
+## CI with Nx
+
+This tutorial walked you through how Nx can improve the developer experience for local development, but Nx can also make a big difference in CI. Without adequate tooling, CI times tend to grow exponentially with the size of the codebase. Nx helps reduce wasted time in CI with the [`affected` command](/ci/features/affected) and Nx Replay's [remote caching](/ci/features/remote-cache). Nx also [efficiently parallelizes tasks across machines](/ci/concepts/parallelization-distribution) with Nx Agents.
+
+To set up Nx Replay run:
+
+```shell
+nx connect
+```
+
+And click the link provided. You'll need to follow the instructions on the website to sign up for your account.
+
+Then you can set up your CI with the following command:
+
+```shell
+nx generate ci-workflow --ci=github
+```
+
+{% callout type="note" title="Choose your CI provider" %}
+You can choose `github`, `circleci`, `azure`, `bitbucket-pipelines`, or `gitlab` for the `ci` flag.
+{% /callout %}
+
+This will create a default CI configuration that sets up Nx Cloud to [use distributed task execution](/ci/features/distribute-task-execution). This automatically runs all tasks on separate machines in parallel wherever possible, without requiring you to manually coordinate copying the output from one machine to another.
+
+## Next Steps
```

Comment:
Maybe we should have a similar section on "Releasing with Nx Release" like we have for CI?

Reply:
Is increasing `nx release` adoption one of our company OKRs? I feel like adding another section between `CI with Nx` and the `Next Steps` will decrease the impact of the call to action.  Some people will forget about CI as the next logical step and go look at `nx release`.  I think we should keep the message consistent and focused on going to CI next.

Reply:
`nx release` is a major feature that closes a gap many ppl were complaining about wrt Nx. And an `npm workspace` is often something where the next step would be to implement a release process of some of these pacakges. We don't have to have a dedicated section but I'd at least mention it in the next steps so ppl have something to follow up with.

Reply:
I'll make a section before the CI section about Nx release.  I agree that is particularly important for the NPM workspace scenario.  I think if release is described first, it won't take away from the ending focus on CI.